runtime/src/system_instruction_processor.rs

```diff
@@ -1103,7 +1125,7 @@ mod tests {
             0,
             &MockInvokeContext::new(vec![]),
         )
-        .is_ok(),);
+        .is_err(),);
```

Comment:
Can we `assert_eq!(transfer(), Err(InstructionError::MissingRequiredSignature));` here instead?  Reordering of errors will break consensus and we should be able to catch it.